clients/ui/frontend/src/app/context/NotificationContext.tsx

```diff
@@ -0,0 +1,82 @@
+import { AlertVariant } from '@patternfly/react-core';
+import React, { createContext } from 'react';
+
+export type Notification = {
+  id?: number;
+  status: AlertVariant;
+  title: string;
+  message?: React.ReactNode;
+  hidden?: boolean;
+  read?: boolean;
+  timestamp: Date;
+};
+
+type NotificationAction =
+  | {
+      type: 'add_notification';
```

Comment:
I would create a new enum to detail the types, it's a nice way to keep everything altogehter a part from the type itself.